src/components/NewHathorAlert.js

```diff
@@ -34,41 +35,80 @@ const NewHathorAlert = forwardRef(({ type, text, showAlert, fixedPosition }, ref
    * @param {number} duration - The display duration for the alert in milliseconds.
    */
   const show = duration => {
-    if (alertDiv.current) {
-      alertDiv.current.classList.add('show');
-      setTimeout(() => {
-        alertDiv.current.classList.remove('show');
-      }, duration);
+    // If the component is not in a fixed position, only handle the alert component
+    if (!fixedPosition) {
+      showAlertComponent();
+      return;
     }
-  };
 
-  useEffect(() => {
-    if (showAlert === undefined) {
+    // No-op if the container is unavailable
+    if (!containerDiv?.current) {
       return;
     }
-    if (showAlert) {
+
+    // The component is in a fixed position: managing its parent container to allow for animations
+    // and precise positioning
+    containerDiv.current.classList.add('show');
+    setTimeout(showAlertComponent, 100); // Delay the alert display to accomodate animations
+    setTimeout(() => {
+      // By the time the timeout is called, the container may have been unmounted
+      if (containerDiv?.current) {
+        containerDiv.current.classList.remove('show');
+      }
+    }, duration + 500);
+
+    /**
+     * Handles the Alert component display and removal
+     */
+    function showAlertComponent() {
+      // No-op if the ref is unavailable
+      if (!alertDiv?.current) {
+        return;
+      }
+
       alertDiv.current.classList.add('show');
-    } else {
-      alertDiv.current.classList.remove('show');
+      setTimeout(() => {
+        // By the time the timeout is called, the component may have been unmounted
+        if (alertDiv?.current) {
+          alertDiv.current.classList.remove('show');
+        }
+      }, duration);
```

Comment:
You don't have this `duration` parameter here, if I'm not wrong. The diffs are a bit confusing but it seems like you should be receiving it as a parameter in the `showAlertComponent` also.

Reply:
This function is actually declare inside the `show()` function, so it shares the `duration` parameter.
The diff visualizarion is confusing, since some of the methods were obsolete and were removed, but you can see it more clearly on the [file itself](https://github.com/HathorNetwork/hathor-explorer/blob/bc594aee5485f6e9c262e63b3c403edee1ec981e/src/components/NewHathorAlert.js#L63-L75).